librhash/byte_order.c

```diff
@@ -74,7 +74,7 @@ unsigned rhash_ctz(unsigned x)
 void rhash_swap_copy_str_to_u32(void* to, int index, const void* from, size_t length)
 {
 	/* if all pointers and length are 32-bits aligned */
-	if ( 0 == (( (int)((char*)to - (char*)0) | ((char*)from - (char*)0) | index | length ) & 3) ) {
+	if ( 0 == ( ((uintptr_t)to & 3) | ((uintptr_t)from & 3) | (index & 3) | (length & 3) ) ) {
```

Comment:
The `& 3` was left outside of OR operands, to optimize the expression by hands and to not depend on compiler optimization. So it should be written as:
```c
if ( 0 == (( (uintptr_t)to | (uintptr_t)from | (uintptr_t)index | length ) & 3) ) {
```